os/workgroup_base.h

```diff
@@ -0,0 +1,78 @@
+#ifndef __OS_WORKGROUP_BASE__
+#define __OS_WORKGROUP_BASE__
+
+#ifndef __OS_WORKGROUP_INDIRECT__
+#error "Please #include <os/workgroup.h> instead of this file directly."
+#endif
+
+#include <sys/types.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <stdbool.h>
+#include <string.h>
+#include <stdlib.h>
+
+#include <mach/port.h>
```

Comment:
Unprotected `mach/port.h` is a portability break. `Availability.h` appears to be missing, and `os/base.h` doesn't exist. `os/object.h` and `os/clock.h` does exist, but `os/clock.h` also refers to `os/base.h`, so something is missing here.

Reply:
Seems like `os/base.h` should switch on `os/generic_*_base.h`. `os/generic_*_basae.h` needs to also #define away SPI_UNAVAILABLE...